### PR TITLE
Feature/19053 add group to print pin card

### PR DESCRIPTION
**Pull request by pris54**
Display group name in pin cards.
Same work as in closed PR #538 (branch needed to be renamed)
NOTE: **PLEASE SEE #538 for a review **

### CODE REVIEW

**Comment by pris54** [comment]
Please see PR #538 for a full review.
Suggestions are:
- Add unit tests for the new argument added to ```getPupilsWithActivePins```
- Refactor ```getPupilsWithActivePins``` so that only one argument is needed

File: admin/controllers/pupil-pin.js [hunk 1]
```diff
@@ -122,14 +122,21 @@ const getGeneratedPinsList = async (req, res, next) => {
...
+ */
 const getPrintPins = async (req, res, next) => {
   res.locals.pageTitle = 'Print pupils'
   let pupils
   let school
   let qrDataURL
   const date = dateService.formatDayAndDate(new Date())
   try {
-    pupils = await pinService.getPupilsWithActivePins(req.user.School)
+    pupils = await pinService.getPupilsWithActivePins(req.user.School, req.user.schoolId)
```
Comment (GuyHarwood) [inline admin/controllers/pupil-pin.js]: as discussed - one identifier only

File: admin/services/pin.service.js [hunk 2]
```diff
@@ -12,12 +13,18 @@ const pinService = {}
 /**
  * Get pupils with active pins
  * @param dfeNumber
- * @returns {Array}
+ * @param schoolId (optional, defaults to null)
+ * @returns {Promise<*>}
  */
-pinService.getPupilsWithActivePins = async (dfeNumber) => {
+pinService.getPupilsWithActivePins = async (dfeNumber, schoolId = null) => {
```
Comment (GuyHarwood) [inline admin/services/pin.service.js]: as discussed, one identifier only

Comment (GuyHarwood) [reply admin/services/pin.service.js]: also need updated tests

**Comment by pris54** [comment]
@GuyHarwood @jon-shipley Please have a look at the new solution. Thanks.

File: admin/services/data-access/pupil.data.service.js [hunk 3]
```diff
@@ -163,9 +163,12 @@ pupilDataService.sqlFindPupilsWithActivePins = async (dfeNumber) => {
   winston.debug(`sqlFindPupilsWithActivePins: called with [${dfeNumber}]`)
```
Comment (GuyHarwood) [inline admin/services/data-access/pupil.data.service.js]: you should be OK to remove this now (unless you need it for further debugging?) as the sql service spits out all queries in debug mode

Comment (pris54) [reply admin/services/data-access/pupil.data.service.js]: @GuyHarwood I didn't write that, I'm assuming whoever wrote it left if for a good reason.

Comment (GuyHarwood) [reply admin/services/data-access/pupil.data.service.js]: unless you need it, please remove it.

**Comment by pris54** [comment]
@GuyHarwood debug line removed.

**Comment by GuyHarwood** [comment]
thanks @pris54

**Pull request merged**